lib/intersperse/index.js

```diff
@@ -0,0 +1,11 @@
+import _curry2 from '../_internal/_curry2'
+
+export default _curry2(function intersperse(separator, list) {
+  var result = []
+  for (var i = 0; i < list.length; i++) {
+    i === list.length - 1
```

Comment:
You can iterate from 0 to n-1 and then just add last element without a condition. It must be faster not to check all the time

Reply:
It have sense, but i've made tests:

whole `for`:
<img width="391" alt="aleksandrbabkin_aleksandrs-macbook-pro____documents_nanoutils__zsh_" src="https://user-images.githubusercontent.com/25172855/37569563-172f27f6-2af5-11e8-9bc2-6d60dcc1d6cc.png">

`for` with no check inside:
<img width="390" alt="intersperse - for no check" src="https://user-images.githubusercontent.com/25172855/37569554-fd283fd2-2af4-11e8-9224-ece747b31a14.png">

Tests have been run for 100 times each with cash clearing

Reply:
![image](https://user-images.githubusercontent.com/4208480/37569610-1c75a5a4-2af6-11e8-86ea-3355d476f73c.png)